### PR TITLE
[chore] [processor/attributes] Remove usage of pcommon.Map.Sort in tests

### PR DESCRIPTION
**Pull request by dmitryax**
Use comparetest module instead.

Updates https://github.com/open-telemetry/opentelemetry-collector/issues/6688

### CODE REVIEW

**Review by TylerHelmuth — APPROVED** [review]

File: processor/attributesprocessor/attributes_metric_test.go [hunk 1]
```diff
@@ -55,91 +54,19 @@ func generateMetricData(resourceName string, attrs map[string]interface{}) pmetr
...
-	case pmetric.MetricTypeSummary:
-		dps := m.Summary().DataPoints()
-		for i := 0; i < dps.Len(); i++ {
-			//nolint:errcheck
-			dps.At(i).Attributes().FromRaw(attrs)
-			dps.At(i).Attributes().Sort()
-		}
+	m.SetName("metric1")
+
+	// TODO: Add tests for other types.
```
Comment (mx-psi) [inline processor/attributesprocessor/attributes_metric_test.go:59]: Can you link to an issue for doing this later?

**Comment by dmitryax** [comment]
Turning it into a draft since the metric tests are not actually fixed. I'll wait for https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/17017 to be merged first

**Comment by dmitryax** [comment]
To unblock removing `Map.Sort(),` I skipped the broken tests that will be fixed by https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/17017

File: processor/attributesprocessor/attributes_metric_test.go (hunk 1 above)
Comment (dmitryax) [reply processor/attributesprocessor/attributes_metric_test.go]: I actually don't think other types are necessary for this processor. So removed the comment

**Pull request merged**